### PR TITLE
feat(ui): add tls_certificate to general slice

### PR DESCRIPTION
**Pull request by Caleb-Ellis**
## Done

- Connect `general.tls_certificate` endpoint to the store

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- N/A

## Fixes

Fixes canonical-web-and-design/app-tribe#818

### CODE REVIEW

File: ui/src/app/store/general/types/base.ts [hunk 1]
```diff
@@ -224,6 +224,16 @@ export type PowerTypesState = {
   loading: boolean;
 };
 
+export type TLSCertificate =
+  | Omit<CertificateData, "private_key"> & CertificateMetadata;
+
+export type TLSCertificateState = {
+  errors: APIError;
+  data: TLSCertificate | EmptyObject | null;
```
Comment (huwshimi) [inline ui/src/app/store/general/types/base.ts:232]: It seems a little unhelpful that it can return an empty object...

**Review by huwshimi — APPROVED** [review]
Looks good. One comment that you can ignore.

File: ui/src/app/store/general/types/base.ts (hunk 1 above)
Comment (Caleb-Ellis) [reply ui/src/app/store/general/types/base.ts:232]: Yep I've mentioned it to Anton so I might be able to remove this in the future, but this is how it is currently

Comment (troyanov) [reply ui/src/app/store/general/types/base.ts:232]: Sorry, I was not aware of what should be returned. It is changed now https://git.launchpad.net/maas/commit/?id=6c6bc1aa1f93184909823677f4251d352d804ce1

Comment (Caleb-Ellis) [reply ui/src/app/store/general/types/base.ts:232]: @troyanov no worries, I should have mentioned this before you started implementing it. Anyway, thanks for addressing it so quick!

**Pull request merged**